api/v3/Order.php

```diff
@@ -127,6 +127,40 @@ function civicrm_api3_order_create(&$params) {
 }
 
 /**
+ * Cancel a Order.
+ *
+ * @param array $params
+ *   Input parameters.
+ *
+ * @return array
+ */
+function civicrm_api3_order_cancel($params) {
+  $contributionStatuses = CRM_Contribute_PseudoConstant::contributionStatus(NULL, 'name');
+  $params['contribution_status_id'] = array_search('Cancelled', $contributionStatuses);
+  $result = civicrm_api3('Contribution', 'create', $params);
+  if (!CRM_Utils_Array::value('is_error', $result)) {
+    CRM_Contribute_BAO_Contribution::transitionComponents($params);
+  }
+  return civicrm_api3_create_success(CRM_Utils_Array::value('values', $result), $params, 'Order', 'cancel');
```

Comment:
CRM_Utils_Array::value('values', $result) should just be $result['values']

$result['values'] should always be returned by all api calls and if not an e-notice is appropriate